homeassistant/components/homematicip_cloud/services.py

```diff
@@ -142,58 +132,61 @@
             await _set_active_climate_profile(hass, service)
 
     hass.services.async_register(
-        HMIPC_DOMAIN,
-        SERVICE_ACTIVATE_ECO_MODE_WITH_DURATION,
-        async_call_hmipc_service,
+        domain=HMIPC_DOMAIN,
+        service=SERVICE_ACTIVATE_ECO_MODE_WITH_DURATION,
+        service_func=async_call_hmipc_service,
         schema=SCHEMA_ACTIVATE_ECO_MODE_WITH_DURATION,
     )
 
     hass.services.async_register(
-        HMIPC_DOMAIN,
-        SERVICE_ACTIVATE_ECO_MODE_WITH_PERIOD,
-        async_call_hmipc_service,
+        domain=HMIPC_DOMAIN,
+        service=SERVICE_ACTIVATE_ECO_MODE_WITH_PERIOD,
+        service_func=async_call_hmipc_service,
         schema=SCHEMA_ACTIVATE_ECO_MODE_WITH_PERIOD,
     )
 
     hass.services.async_register(
-        HMIPC_DOMAIN,
-        SERVICE_ACTIVATE_VACATION,
-        async_call_hmipc_service,
+        domain=HMIPC_DOMAIN,
+        service=SERVICE_ACTIVATE_VACATION,
+        service_func=async_call_hmipc_service,
         schema=SCHEMA_ACTIVATE_VACATION,
     )
 
     hass.services.async_register(
-        HMIPC_DOMAIN,
-        SERVICE_DEACTIVATE_ECO_MODE,
-        async_call_hmipc_service,
+        domain=HMIPC_DOMAIN,
+        service=SERVICE_DEACTIVATE_ECO_MODE,
+        service_func=async_call_hmipc_service,
         schema=SCHEMA_DEACTIVATE_ECO_MODE,
     )
 
     hass.services.async_register(
-        HMIPC_DOMAIN,
-        SERVICE_DEACTIVATE_VACATION,
-        async_call_hmipc_service,
+        domain=HMIPC_DOMAIN,
+        service=SERVICE_DEACTIVATE_VACATION,
+        service_func=async_call_hmipc_service,
         schema=SCHEMA_DEACTIVATE_VACATION,
     )
 
-    hass.services.async_register(
-        HMIPC_DOMAIN,
-        SERVICE_SET_ACTIVE_CLIMATE_PROFILE,
-        async_call_hmipc_service,
+    async_register_admin_service(
+        hass=hass,
+        domain=HMIPC_DOMAIN,
+        service=SERVICE_SET_ACTIVE_CLIMATE_PROFILE,
```

Comment:
(note: I'm not familiar with homematic)

Isn't this something like a climate preset ? Should this be an admin service or should it use the service helper `verify_domain_control` where they can call this service if they can control any homematic device?

Reply:
Good point. I'll add verify_domain_control and make this service user accessible.